Add Rewards integration to Safety Module

EncodePanda opened the pull request:
# Feature

Integrate the safety module with the rewards program

# Demo
https://github.com/user-attachments/assets/e2fd3e65-5e87-4518-9dce-52b3a161781f

EncodePanda committed:
Add config and rename safetyModuleConfig

EncodePanda committed:
Add rewards staking in the Safety Module

EncodePanda committed:
Change the visibility conditions to check userSharesTotal instead of userShares

AdnanSlef reviewed (changes requested):
checking the "stake" checkbox when depositing to the safety module should result in your staked sUSDST stat increasing rather than your unstaked stat increasing, and when you go to withdraw the max should be lower than your total deposited until you check the "include staked" box, whereupon it should increase. these were not observed in testing.

EncodePanda committed:
Introduce waitForBalanceUpdate that waits for the balance update in the indexer

Both lending.service and safety.service query Cirrus immediately after
transaction was executed on chain.

This means that in some rare conditions, the indexer is not updated on time and
the resulted lpTokenAfter is the same as lpTokenBefore, resulting in 0 tokens to
be staked for rewards.

The waitForBalanceUpdate is the simplest way to deal with this race condition.

EncodePanda committed:
Modify toast messages depending on the stakeSUSDST checkbox
